pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -459,36 +462,18 @@ func (r *NetworkInfoReconciler) CollectGarbage(ctx context.Context) {
 	}
 }
 
-func (r *NetworkInfoReconciler) fetchStaleVPCsByNamespace(ctx context.Context, ns string) ([]*model.Vpc, error) {
-	isShared, err := r.Service.IsSharedVPCNamespaceByNS(ctx, ns)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// if the Namespace has been deleted, we never know whether it`s a shared Namespace, The GC will delete the stale VPCs
-			log.Info("Namespace does not exist while fetching stale VPCs", "Namespace", ns)
-			return nil, nil
-		}
-		return nil, fmt.Errorf("failed to check if Namespace is shared for NS %s: %w", ns, err)
-	}
-	if isShared {
-		log.Info("Shared Namespace, skipping deletion of NSX VPC", "Namespace", ns)
-		return nil, nil
+func (r *NetworkInfoReconciler) deleteVPCsByName(ctx context.Context, ns string) error {
+	staleVPCs := r.Service.GetVPCsByNamespace(ns)
```

Comment:
I see the major change compared with previous is that:
We don't check if this namespace is sharedVPC namespace or not.
So, we always assume the requested namespace in deleteVPCsByName which has a networkinfo CR and this CR has been deleted, so we need to delete the VPC only for this VPC in this namespace, is it?

Reply:
No, the change is to resolve the bug that, when a NetworkInfo CR is deleted, its Namespace may not remove from apiserver. However, as long as the Namespace is desired to delete, we should delete the corresponding NSX VPC. The original code doesn't correctly check the Namespace is desired to delete or not, instead, it thinks the NS as valid as long as it exists in the apiserver. So we actually do not delete the NSX VPC even if the user expect to delete NS, which caused many stale resources left on NSX before next round of GC.